lib/BridgedRoom.js

```diff
@@ -137,18 +187,28 @@ BridgedRoom.prototype.toEntry = function() {
 };
 
 BridgedRoom.prototype.onMatrixMessage = function(message) {
-    if (!this._slack_webhook_uri) return Promise.resolve();
+    if (!this._slack_webhook_uri && !this._slack_bot_token) return Promise.resolve();
 
     return this._main.getOrCreateMatrixUser(message.user_id).then((user) => {
         var body = substitutions.matrixToSlack(message, this._main);
 
+        var uri = (this._slack_bot_token) ? "https://slack.com/api/chat.postMessage" : this._slack_webhook_uri;
+
         var sendMessageParams = {
             method: "POST",
             json: true,
-            uri: this._slack_webhook_uri,
-            body: body
+            uri: uri,
+            body: body,
         };
 
+        if (this._slack_bot_token) {
+            sendMessageParams.headers = {
+                Authorization: 'Bearer ' + this._slack_bot_token
+            };
+            sendMessageParams.body.as_user = false;
+            sendMessageParams.body.channel = this._slack_channel_id;
+        }
+
         sendMessageParams.body.username = user.getDisplaynameForRoom(message.room_id);
```

Comment:
Actually I am wondering if this will still work with a bot user. See here the part about "Authoring messages": https://api.slack.com/changelog/2017-09-the-one-about-usernames#changing.

If I am correct, a bot user cannot change the username anymore :(

Reply:
I did some reading and this page seemed to suggest you could: https://api.slack.com/methods/chat.postMessage (I have no idea just providing links 😆 )

Reply:
Hmm, it looks like maybe they are phasing out that support, which would be disasterous for this AS, both in webhook and bot mode.

Reply:
The username & avatar changes definitely work as of now, not sure what will happen in the future.